packages/source-iotsitewise/src/initialize.ts

```diff
@@ -11,22 +11,22 @@ import {
   SiteWiseAssetTreeSession,
 } from './asset-modules';
 import { SiteWiseComponentSession } from './component-session';
-import { sitewiseSdk } from './sitewise-sdk';
+import { getSiteWiseClient } from './sitewise-sdk';
+import { getIotEventsClient } from './events-sdk';
 import { createSiteWiseAssetDataSource } from './asset-modules/asset-data-source';
 import { createDataSource } from './time-series-data';
 import { Credentials, Provider as AWSCredentialsProvider } from '@aws-sdk/types';
 import { IoTSiteWiseClient } from '@aws-sdk/client-iotsitewise';
+import { IoTEventsClient } from '@aws-sdk/client-iot-events';
 import { assetSession } from './sessions';
+import { SiteWiseAlarmModule } from './alarms/iotevents';
 
-type SiteWiseDataSourceInitInputs = (
-  | {
-      iotSiteWiseClient: IoTSiteWiseClient;
-    }
-  | {
-      awsCredentials: Credentials | AWSCredentialsProvider<Credentials>;
-      awsRegion: string;
-    }
-) & {
+export type SiteWiseDataSourceInitInputs = {
+  registerDataSources?: boolean;
+  iotSiteWiseClient?: IoTSiteWiseClient;
+  iotEventsClient?: IoTEventsClient;
+  awsCredentials?: Credentials | AWSCredentialsProvider<Credentials>;
+  awsRegion?: string;
```

Comment:
do we need validation for passing in credentials without a region? seems like the semantics of this type definition has changed

Reply:
We set a default region for the sdks.